src/Basis/BasisManager.cpp

```diff
@@ -42,37 +42,34 @@ std::vector<Base> BasisManager::getBasisList(Source s) {
 	return this->bases;
 }
 
-Base BasisManager::Builder::build() {
+Base BasisManager::Builder::build(int dimension) {
 	//TODO: Eventually add controls...
-    int dimension = 0;
-
 	return Base(Base::basePreset::Custom, dimension, c_base, d_base);
 }
 
 Base BasisManager::Builder::build(Base::basePreset b, int dimension) {
     //TODO: Eventually add controls...
-        return Base(b, dimension, c_base, d_base);
-    }
+    return Base(b, dimension, c_base, d_base);
+}
 
 // --- Factory Methods --- //
 Base BasisManager::Builder::build(Base::basePreset b, int dimension, double mesh, int nbox) {
     //TODO: Eventually add controls...
 
     switch(b){
-
-        case Base::basePreset::Custom: throw std::invalid_argument("Custom basis not meaningful with parameters!");
-            break;
         case Base::basePreset::Cartesian:
             std::cout<< "Building Cartesian basis in " << dimension << "dimensions, with nbox = " << nbox << ", mesh = " << mesh << std::endl;
-            for(int i = 0; i < dimension; i++)
-            {
+            for(int i = 0; i < dimension; i++) {
                 this->addContinuous(mesh, nbox);
             }
             break;
+
         // todo: find criterium criterium to define Lmax given only r. Harmonic oscillator r/l relation?
```

Comment:
It may sound like bullshit, but comment keywords should go uppercase (e.g. TODO, TOCLEAN etc...) for best IDE compatibility

Reply:
Right!